target/linux/ath79/dts/qca9533_tplink_tl-wr841hp-v3.dts

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca953x.dtsi"
+
+/ {
+	compatible = "tplink,tl-wr841hp-v3", "qca,qca9533";
+	model = "TP-Link TL-WR841HP v3";
+
+	aliases {
+		label-mac-device = &wmac;
+		led-boot = &led_pwr;
+		led-failsafe = &led_pwr;
+		led-running = &led_pwr;
+		led-upgrade = &led_pwr;
+	};
+
+	leds: leds {
+		compatible = "gpio-leds";
+
+		wifi {
+			label = "tp-link:green:wlan";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		led_pwr: pwr {
+			label = "tp-link:green:pwr";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		wan {
+			label = "tp-link:green:wan";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+		};
+
+		wan_red {
+			label = "tp-link:red:wan";
+			gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+		};
+
+		lan {
+			label = "tp-link:green:lan";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+
+		wps {
+			label = "tp-link:green:wps";
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		rfkill {
+			label = "RFKILL button";
+			linux,code = <KEY_RFKILL>;
+			gpios = <&gpio 0 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		wps {
+			label = "WPS button";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+};
+
+&uart {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+	num-cs = <1>;
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			uboot: partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x020000>;
+				read-only;
+			};
+
+			partition@20000 {
+				compatible = "tplink,firmware";
+				label = "firmware";
+				reg = <0x020000 0x7d0000>;
+			};
+
+			art: partition@7f0000 {
+				label = "art";
+				reg = <0x7f0000 0x010000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&eth1 {
+	mtd-mac-address = <&uboot 0x1fc00>;
+};
+
+&eth0 {
+	status = "okay";
+	phy-handle = <&swphy4>;
+	mtd-mac-address = <&uboot 0x1fc00>;
+	mtd-mac-address-increment = <1>;
```

Comment:
Is the MAC address assignment chosen in accordance to vendor firmware? If yes, I'd be happy about a short note/overview in the commit message (effectively showing the increment of WAN/LAN/Wifi with respect to label MAC address and the fact that you checked it against vendor settings)

Reply:
Ok, I will flash back to vendor firmware & check again.

Reply:
If you checked, you don't need to check _again_. But if you didn't check, you should check.

Reply:
I'm checked my router u-boot offset at 0x1fc00, it's contain exactly label mac-address.
On vendor, label mac-addres=lan/wifi mac-address. wan mac-address=lan mac-address +1
example my router: lan/wifi mac=D4:6E:0E:XX:XX:EA ; wan mac =D4:6E:0E:XX:XX:EB
The same for openwrt firmware.

Reply:
Okay, thanks. Then please add these findings to the commit message as well (you might just put this in a short table or something):

```
MAC addresses based on vendor firmware:
LAN    *:ea    label
WAN    *:eb   label +1
2.4 GHz  *:ea   label

The label MAC address in found in u-boot 0x1fc00.
```

Reply:
I committed before you said, how to fix it?

Reply:
You will have to squash all your changes into one proper commit anyway (use git rebase -i ...). Then, force-push them here again (git push -f ...). Despite, please make formatting nicer than in my quickly typed example.